tests/integration/targets/ipam_address/tasks/main.yml

```
@@ -0,0 +1,219 @@
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+      register: address
+    - name: Get information about the Address
+      infoblox.bloxone.ipam_address_info:
+        filters:
+          address: "10.0.0.3"
+          space: "{{ _ip_space.id }}"
+      register: address_info
+    - assert:
+        that:
+          - address is changed
+          - address is not failed
+          - address_info.objects | length == 1
+          - address_info.objects[0].id == address.id
+
+    - name: Create an Address(idempotent)
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+      register: address
+    - assert:
+        that:
+          - address is not changed
+          - address is not failed
+
+    - name: Delete an Address (check mode)
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+      check_mode: true
+      register: address
+    - name: Get information about the Address
+      infoblox.bloxone.ipam_address_info:
+        filters:
+          address: "10.0.0.3"
+          space: "{{ _ip_space.id }}"
+      register: address_info
+    - assert:
+        that:
+          - address is changed
+          - address is not failed
+          - address_info.objects | length == 1
+
+    - name: Delete an Address
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+      register: address
+    - name: Get information about the Address
+      infoblox.bloxone.ipam_address_info:
+        filters:
+          address: "10.0.0.3"
+          space: "{{ _ip_space.id }}"
+      register: address_info
+    - assert:
+        that:
+          - address is changed
+          - address is not failed
+          - address_info.objects | length == 0
+
+    - name: Delete an Address(idempotent)
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        space: "{{ _ip_space.id }}"
+        state: "absent"
+      register: address
+    - assert:
+        that:
+          - address is not changed
+          - address is not failed
+
+    - name: Create an Address with Name
+      infoblox.bloxone.ipam_address:
+        address: "10.0.0.3"
+        names:
+          - name: "test-1"
+            type: "user"
+        space: "{{ _ip_space.id }}"
+        state: "present"
+      register: address
+    - name: Get information about the Address
+      infoblox.bloxone.ipam_address_info:
+        filters:
+          address: "10.0.0.3"
+          space: "{{ _ip_space.id }}"
+      register: address_info
+    - assert:
+        that:
+          - address is changed
+          - address is not failed
+          - address_info.objects | length == 1
+          - address_info.objects[0].names[0].name == address.object.names[0].name
```

Use actual values on the Right hand hand side instead of values from address variable . 
This is supposed to be implemented for all UTs. 

```suggestion
          - address_info.objects[0].names[0].name == "test-1"
          - address_info.objects[0].names[0].type == "user"
```